### PR TITLE
Handle updates to clusterDeployment name and namespace

### PR DESCRIPTION
**Pull request by sacharya**
When clusterDeployment name is updated, the reference in syncset is updated
and propagation is handled by syncset. Since a syncset is always created in
the namespace of the clusterDeployment, anytime namespace is changed,
syncset needs to be deleted from old namespace and recreated in the new
namespace.

### CODE REVIEW

File: pkg/controller/baremetalasset/baremetalasset_controller.go [hunk 1]
```diff
@@ -679,6 +649,62 @@ func (r *ReconcileBareMetalAsset) deleteSyncSet(instance *midasv1alpha1.BareMeta
...
+		return err
+	}
+
+	if len(found.Items) > 0 {
+		err = fmt.Errorf("Found SyncSetInstances in namespace: %v with label %v:%v. Expected: (%v) Actual: (%v)", hscRef.Namespace, hiveconstants.SyncSetNameLabel, hscRef.Name, 0, len(found.Items))
+		return err
+	}
+
+	// Remove SyncSet from related objects
+	objectreferencesv1.RemoveObjectReference(&instance.Status.RelatedObjects, hscRef)
```
Comment (rwsu) [inline pkg/controller/baremetalasset/baremetalasset_controller.go]: Add a check for error from RemoveObjectReference.

Comment (sacharya) [reply pkg/controller/baremetalasset/baremetalasset_controller.go]: Done

File: pkg/controller/baremetalasset/baremetalasset_controller.go [hunk 2]
```diff
@@ -679,6 +649,62 @@ func (r *ReconcileBareMetalAsset) deleteSyncSet(instance *midasv1alpha1.BareMeta
...
+	reqLogger.Info("Cleaning up Hive SyncSet", "SyncSet.Name", hscRef.Name, "SyncSet.Namespace", hscRef.Namespace)
+	err := r.client.Delete(context.TODO(), &hivev1.SyncSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: hscRef.Namespace,
+			Name:      hscRef.Name,
+		},
+	})
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			reqLogger.Error(err, "Failed to delete Hive SyncSet")
```
[the remark this answers is not in the document]
Comment (sacharya) [reply pkg/controller/baremetalasset/baremetalasset_controller.go]: Done

File: pkg/controller/baremetalasset/baremetalasset_controller.go [hunk 3]
```diff
@@ -679,6 +649,62 @@ func (r *ReconcileBareMetalAsset) deleteSyncSet(instance *midasv1alpha1.BareMeta
...
+			break
+		}
+	}
+	if hscRef == (corev1.ObjectReference{}) {
+		// Nothing to do if no such syncset was found
+		return nil
+	}
+
+	// Delete syncset in old namespace
+	reqLogger.Info("Cleaning up Hive SyncSet", "SyncSet.Name", hscRef.Name, "SyncSet.Namespace", hscRef.Namespace)
```
[the remark this answers is not in the document]
Comment (sacharya) [reply pkg/controller/baremetalasset/baremetalasset_controller.go]: Done

**Pull request merged**